### PR TITLE
gh-79: better lensing efficiency

### PR DESCRIPTION
**Pull request by marcobonici**
In this PR I have added a fast method for the evaluation of the lensing efficiency, based on a triangular simpson matrix.
The old functions to compute it were not removed, but I simply added a suffix `_check` at the end of the old functions name.
Once we are gonna have unit tests, these can be used to test the accuracy of the new function.

Can you please check and review it, @gcanasherrera ?

### CODE REVIEW

**Comment by gcanasherrera** [comment]
@marcobonici Can I start reviewing?

File: cloelite/auxiliary/math_utils.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import jax.numpy as jnp
 
 # This will go to a math-utils module, for the moment stays here
-max_size = 300 
+max_size = 300
```
Comment (gcanasherrera) [inline cloelite/auxiliary/math_utils.py:5]: Put it already in math utils, we will re-arrange math utils in #74

File: cloelite/observables/photo.py [hunk 2]
```diff
@@ -170,14 +171,29 @@ def get_lensing_efficiency_bin(self, z, bin_idx):
...
         n_bins = self.dndz.shape[0]
         efficiency = self.get_lensing_efficiency_bin(z, 0)
         for i in np.arange(1,n_bins):
             efficiency = np.vstack([efficiency, self.get_lensing_efficiency_bin(z, i)])
 
         return efficiency
 
+    def get_lensing_efficiency(self, z):
+        dndz = self.dndz
+        dz = z[1]-z[0]#assuming equispaced!
```
Comment (gcanasherrera) [inline cloelite/observables/photo.py:184]: We need to start collecting this assumptions somewhere. Open a task about generalising this (so we don't lose track), please

File: cloelite/observables/photo.py [hunk 3]
```diff
@@ -200,6 +216,22 @@ def get_window(self, z):
         """
         return self.get_lensing_window(z) + self.get_window_IA(z)
 
+    def get_window_new(self, z):
```
Comment (gcanasherrera) [inline cloelite/observables/photo.py]: I'm not sure I agree with the names used. Let's give some descriptive names instead of _new.

**Comment by marcobonici** [comment]
> @marcobonici Can I start reviewing?

Sure.

**Review by gcanasherrera — COMMENTED** [review]
All for you to go!

File: cloelite/observables/photo.py (hunk 3 above)
Comment (marcobonici) [reply cloelite/observables/photo.py]: I do not agree with the name chosen as well. I will modify it.
